src/app/components/task/task.component.html

```diff
@@ -12,4 +12,14 @@
             <p>{{task.date.toLocaleDateString('fr-FR', { month: 'long', day: 'numeric' })}}</p>
         </div>
     </div>
-</div>
\ No newline at end of file
+    <button class="task-delete-button" (click)="openDeleteTaskModal()">X</button>
+</div>
+
+<app-modal id="add-task-modal" *ngIf="isDeleteModalOpen" (closeModal)="closeDeleteTaskModal()">
```

Comment:
you can create a single toggleDelete() method and call it to open and close the modal